From 79e210753c035d51cd0f28ef65c1bb1463559303 Mon Sep 17 00:00:00 2001
From: Dor Askayo <dor.askayo@gmail.com>
Date: Tue, 12 May 2020 22:51:50 +0300
Subject: [PATCH 03/18] clutter/stage: don't redraw the stage when it can't be
 redrawn

It should be fine not to draw during a stage update, because a new
update would be scheduled by the master clock when a redraw remains
pending for that stage.

https://gitlab.gnome.org/GNOME/mutter/-/merge_requests/1241
---
 clutter/clutter/clutter-stage.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/clutter/clutter/clutter-stage.c b/clutter/clutter/clutter-stage.c
index 61737abd4..2a257c2e4 100644
--- a/clutter/clutter/clutter-stage.c
+++ b/clutter/clutter/clutter-stage.c
@@ -1510,7 +1510,7 @@ _clutter_stage_do_update (ClutterStage *stage)
 
   COGL_TRACE_END (ClutterStageRelayout);
 
-  if (!priv->redraw_pending)
+  if (!priv->redraw_pending || !_clutter_stage_window_can_redraw (priv->impl))
     return FALSE;
 
   if (stage_was_relayout)
-- 
2.26.2

